app/src/main/java/org/dharmaseed/android/PlayTalkActivity.java

```diff
@@ -49,59 +44,59 @@
 import android.widget.TextView;
 import android.widget.Toast;
 
+import androidx.appcompat.app.ActionBar;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.ContextCompat;
+import androidx.localbroadcastmanager.content.LocalBroadcastManager;
+import androidx.media3.common.MediaItem;
+import androidx.media3.common.Player;
+import androidx.media3.session.MediaController;
+import androidx.media3.session.SessionToken;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
 import java.io.FileInputStream;
 import java.io.FileNotFoundException;
 import java.text.DateFormat;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
+import java.util.concurrent.ExecutionException;
 import java.util.GregorianCalendar;
 import java.util.Timer;
 
 public class PlayTalkActivity extends AppCompatActivity
         implements SeekBar.OnSeekBarChangeListener, DeleteTalkFragment.DeleteTalkListener {
 
-    TalkPlayerFragment talkPlayerFragment;
     int talkID;
     DBManager dbManager;
     boolean userDraggingSeekBar;
 
+    boolean shouldSeekToResumePos;
+    int resumePos;
+
     Timer timer;
 
-    private static Talk talk;
+    MediaController mediaController;
+
+    Talk talk;
 
-    private static final String LOG_TAG = "PlayTalkActivity";
+    static final String LOG_TAG = "PlayTalkActivity";
 
     protected static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss";
 
     // request code for writing external storage (the number is arbitrary)
-    private static final int PERMISSIONS_WRITE_EXTERNAL_STORAGE = 9087;
-
-    protected void prepareTalkPlayerFragment()
-    {
-        if (talkPlayerFragment == null)
-            Log.e(LOG_TAG, "talkPlayerFragment must be non-null!");
-
-        final MediaPlayer mediaPlayer = talkPlayerFragment.getMediaPlayer();
-        try {
-            mediaPlayer.reset();
-            if (talk.isDownloaded()) {
-                Log.d(LOG_TAG, "Playing from " + talk.getPath());
-                mediaPlayer.setDataSource("file://" + talk.getPath());
-            } else {
-                mediaPlayer.setDataSource(talk.getAudioUrl());
-            }
-        } catch (java.io.IOException e) {
-            e.printStackTrace();
-            Log.e(LOG_TAG, e.toString());
-        }
-        Log.i(LOG_TAG,"preparing media player for "+talkID);
-        mediaPlayer.prepareAsync();
-    }
+    static final int PERMISSIONS_WRITE_EXTERNAL_STORAGE = 9087;
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_play_talk);
+    }
+
+    @Override
+    protected void onResume() {
```

Comment:
A lot of initialization for this activity was moved to `onResume` because we want to make sure the UI controls are properly synced with the state of the background media player service every time the `PlayTalkActivity` is resumed. Previously, it didn't matter because this activity was the only place we did media playback from, so it wasn't possible to get out of sync.